### ISSUE TITLE
VIEWLIABILITIES - DATE_FROM_INVALID

### ISSUE THREAD

**Issue by davidpipe58** [hmrc/vat-api#363]

**Comment by davidpipe58** [comment]
I'm submitting a Support request and maybe a Bug - API.

We have received our production credentials and our company is now signed up for the public pilot so I am doing some "live" testing.

Our last VAT quarter ended on 30th November so querying Obligations returns two items - the quarter just ended and the quarter just started. We haven't submitted our VAT100 return yet so both are outstanding.

We therefore have no liabilities or payments

Querying Payments returns NOT_FOUND (unless I use a date greater than today, in which case it returns INVALID_TO_DATE).

Querying Liabilities returns DATE_FROM_INVALID in all cases. I have tried various "from" dates and all have the same results.

Is this because there are no liabilities, and if that is the case, should it not be returning NOT_FOUND?

**Comment by narendravijayarao** [comment]
Hi,

For any production related issues please contact SDST at SDSTeam@hmrc.gsi.gov.uk with details, VRN, request you made, and the date and time of the request.

Thanks,

**Comment by MarkAKelly** [comment]
Hi @davidpipe58,

Just as a follow up to the above - for us to best assist you and find your requests in our logs to work out what's happening, we will need a bit of information about your request (date ranges used, time of the request, the request made, VRN used etc) which we can't ask for over public channels.

If you can provide SDST with that information we can have a look into these for you.

The first scenario with payments makes sense.  If there are no payments then NOT_FOUND should be returned.  The INVALID_DATE_TO will be returned for a 'to' date supplied past the current date.

The second scenario we'll need the information to look at.  The DATE_FROM_INVALID error should be returned if the date supplied is before the official MTD date, which has been set (by HMRC) as 1-12-2017.  It will also be returned if the date is malformed, i.e. not in the format of 'yyyy-mm-dd'.  If you could supply SDST with your request info then we can have a proper look for you.

Thanks,

Mark

**Comment by davidpipe58** [comment]
Hi Mark,

I will get this information to you as soon as I can.

Unfortunately we are moving offices tomorrow so I will be shutting down our IT networks shortly. Hopefully I will get it sent over on Monday.

Regards

David Pipe

**Comment by davidpipe58** [comment]
Hi Mark,

Before I started sending you information to look into I thought I better go back and check this out for myself again.

It turns out that the “canned” test dates - from=2017-04-05&to=2017-12-21 - had been left in one of the lower levels of our own API for liabilities.

I hope this hasn’t wasted too much of your time, and thank you for responding to the issue I raised. I will close the issue off on github.

Kind regards

David Pipe
